src/compiler/scanner.ts

```diff
@@ -2887,7 +2903,7 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                         return "\\";
                     }
                     pos--;
-                    return scanEscapeSequence(/*shouldEmitInvalidEscapeError*/ unicodeMode, /*isRegularExpression*/ annexB ? "annex-b" : true);
+                    return scanEscapeSequence(/*shouldEmitInvalidEscapeError*/ unicodeMode, /*isRegularExpression*/ anyUnicodeModeOrNonAnnexB || "annex-b");

```

Comment:
Since `"annexB"` is no longer used in `scanEscapeSequence`, this can just be `true`:
```suggestion
                    return scanEscapeSequence(/*shouldEmitInvalidEscapeError*/ unicodeMode, /*isRegularExpression*/ true);
```